server/src/main/java/io/crate/replication/logical/action/TransportDropSubscriptionAction.java

```diff
@@ -0,0 +1,96 @@
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.ClusterStateTaskExecutor;
+import org.elasticsearch.cluster.block.ClusterBlockException;
+import org.elasticsearch.cluster.block.ClusterBlockLevel;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.metadata.Metadata;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.inject.Singleton;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+
+@Singleton
+public class TransportDropSubscriptionAction extends AbstractDDLTransportAction<DropSubscriptionRequest, AcknowledgedResponse> {
+
+    public static final String ACTION_NAME = "internal:crate:replication/logical/subscription/drop";
+
+    @Inject
+    public TransportDropSubscriptionAction(TransportService transportService,
+                                           ClusterService clusterService,
+                                           ThreadPool threadPool,
+                                           IndexNameExpressionResolver indexNameExpressionResolver) {
+        super(ACTION_NAME,
+            transportService,
+            clusterService,
+            threadPool,
+            indexNameExpressionResolver,
+            DropSubscriptionRequest::new,
+            AcknowledgedResponse::new,
+            AcknowledgedResponse::new,
+            "drop-subscription");
+    }
+
+    @Override
+    public ClusterStateTaskExecutor<DropSubscriptionRequest> clusterStateTaskExecutor(DropSubscriptionRequest request) {
+        return new DDLClusterStateTaskExecutor<>() {
+            @Override
+            protected ClusterState execute(ClusterState currentState, DropSubscriptionRequest dropSubscriptionRequest) throws Exception {
+                Metadata currentMetadata = currentState.metadata();
+                Metadata.Builder mdBuilder = Metadata.builder(currentMetadata);
+
+                SubscriptionsMetadata oldMetadata = (SubscriptionsMetadata) mdBuilder.getCustom(SubscriptionsMetadata.TYPE);
+                if (oldMetadata == null && request.ifExists() == false) {
+                    throw new SubscriptionUnknownException(request.name());
+                } else if (oldMetadata != null) {
+                    if (oldMetadata.subscription().containsKey(request.name())) {
+
+                        SubscriptionsMetadata newMetadata = SubscriptionsMetadata.newInstance(oldMetadata);
+                        newMetadata.subscription().remove(request.name());
+                        assert !newMetadata.equals(oldMetadata) : "must not be equal to guarantee the cluster change action";
+                        mdBuilder.putCustom(SubscriptionsMetadata.TYPE, newMetadata);
+
+                        return ClusterState.builder(currentState).metadata(mdBuilder).build();
```

Comment:
Maybe add a comment that `LogicalReplicationService` will react to cluster change events and take care of stopping the replication.